### PR TITLE
Reset _format_key_to_idx hash properly and avoid check with autovivified hash

### PR DESCRIPTION
**Pull request by dufeiyu**
This PR is to fix the bug. This is causing $entry->sample_field not returned correctly if add_format_field first.

### CODE REVIEW

File: lib/perl/Genome/File/Vcf/Entry.pm [hunk 1]
```diff
@@ -592,7 +592,8 @@ sub add_format_field {
     else {
         my $idx = scalar @{$self->{_format}};
         push @{$self->{_format}}, $field;
-        $self->{_format_key_to_idx}{$field} = $idx;
+        my $format_idx = $self->format_field_index;
+        $self->{_format_key_to_idx} = {%$format_idx, $field => $idx};
```
Comment (ebelter) [inline lib/perl/Genome/File/Vcf/Entry.pm]: Could this be more understandable written as:
```
$format_idx->{ $field } = $idx;
 $self->{_format_key_to_idx} = $format_idx;
```
It also avoids copying the hash into another when using %$format_idx.

Comment (dufeiyu) [inline lib/perl/Genome/File/Vcf/Entry.pm]: makes sense. I will make the change.

File: lib/perl/Genome/File/Vcf/Entry.pm [hunk 2]
```diff
@@ -622,7 +622,9 @@ an example.
 
 sub format_field_index {
     my ($self, $key) = @_;
-    if (!exists $self->{_format_key_to_idx}) {
+    my $format_idx = $self->{_format_key_to_idx};
+
+    unless (scalar keys %$format_idx) {
```
Comment (dufeiyu) [inline lib/perl/Genome/File/Vcf/Entry.pm]: Makes sense. I will make the change.

**Comment by dufeiyu** [comment]
@tmooney  Can you review this too ?  This is the one to fix the bug in Genome::File::Vcf::Entry that we discussed in https://github.com/genome/genome/pull/1582    thanks

**Pull request merged**